graphql/resolve/mutation_rewriter.go

```diff
@@ -1756,8 +1823,19 @@ func existenceQueries(
 
 					// Add the corresponding existence query. As this is the first time we have
 					// encountered this variable, the query is added only once per variable.
-					query := checkXIDExistsQuery(variable, xidString, xid.Name(), typ)
+					query := checkXIDExistsQuery(variable, xidString, xid.Name(), typ, nil)
 					ret = append(ret, query)
+
+					// Add one more existence query if given xid field is inherited from interface and has
+					// unique argument set. This is added to ensure that this xid is unique across all the
+					// implementation of the interface.
+					interfaceTypDef, varInterface := interfaceVariable(typ, varGen,
+						xid.Name(), xidString)
+					if interfaceTypDef != nil {
+						queryInterface := checkXIDExistsQuery(varInterface, xidString, xid.Name(),
+							typ, interfaceTypDef)
+						ret = append(ret, queryInterface)
+					}
```

Comment:
As discussed offline in a call, we will have to handle case of duplicate XID for different implementing types using the variableObjMap or add a TODO